testing/btest/language/vector-any-append.zeek

```diff
@@ -19,4 +31,35 @@ event zeek_init()
 	append(v);
 	append(v);
 	print v;
+
+	local v2: vector of any;
+	v2 += |v|;
+	v2 += v;
+	print v2;
```

Comment:
Reproducing without baseline for illustration:
```
local v: vector of count = vector(1,2,3,4);
local v2: vector of any;
v2 += |v|
v2 += v;
assert v2 === [4, 0, 1, 2, 3];
```
This somehow doesn't feel right to me. The crux being that there's a `vector of any` on the LHS and so I interpret the RHS as a single element of type `any`, expecting `[4, [0,1,2,3]]`.

Think your proposal in #3191 was more about the RHS treatment to enable more of `add-every-element`. But, what if `+=` on a  `vector of any` is a special case and behaves as `add-single-element`? That would align more with prior Zeek 4.2's `vector of any` behavior and fix #3191. `any` is special `+=` on vectors is special and the above seems easier for me to explain (and happy to pick up the ticket from here).

That said, I believe I'm getting in the way and maybe @rsmmr / @ckreibich / @bbannier may have a look for separate intuition/idea.

Reply:
Since I was pinged, to me this behavior seems to be a natural extension with how vectors over non-`any` types behave, e.g.,

```zeek
local x1 = vector(0, 1, );
local x2 = vector(2, 3, );
x1 += x2;
assert(all_set(x1 == vector(0, 1, 2, 3, )));
```

The added caveat in the snippet you point out is that with this patch a `vector of any` and any other vector (e.g., the `vector of count` here) are implicitly covariant (since `any` and any other type are covariant). We are not doing anything with the `any` values in the `vector` here, so I think this should be a consistent use of fine (_The only operation allowed on a variable of type any is assignment._). IMO the semantics are unfortunately still too subtle.

Without knowing any of the history around this, my gut reaction to the special behavior of `vector`'s `+=` if vector element type and RHS type are covariant (i.e., that one can add a `count` value to a `vector of count` with `+=`) would be that it feels strange and might be unnecessary, e.g., `set` does not have this and can only have a compatible `set` on the RHS of `+=` (there's a dedicated `set` operator `add` to add single elements). If there was a discussion for adding `vector` `+=` today my argument would probably be to instead force the user to potentially create a RHS `vector` and handle that efficiently, e.g.,

```zeek
local xs: vector of count;
# xs += 1; # REJECTED: incompatible LHS and RHS.
xs += vector(1); # Works.
```

Reply:
I'm resolving because after pondering away from the computer I'm back at arguing based on the 4.2 behavior.